event/event.go

```diff
@@ -0,0 +1,52 @@
+// Package event provides functionality for publish/suscribe of events.
+package event
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/birdie-ai/golibs/tracing"
+	"gocloud.dev/pubsub"
+)
+
+// Publisher represents a publisher of events of type T.
+// The publisher guarantees that the events conform to our basic schema for events.
+type Publisher[T any] struct {
+	name  string
+	topic *pubsub.Topic
+}
+
+// Body represents the general structure of the body of events.
+type Body[T any] struct {
```

Comment:
Wondering if we need generics here. We'll call json.Marshal over the event, right? so interface{} / any wouldn't be enough? Can we have any guarantees using T?

Reply:
You have the guarantee that you won't send on the same publisher/topic different/arbitrary data. Once built for type T it will only send type T.

Reply:
Of course you still could create multiple publishers with different types with the same topic, but that is more explicit. Send on an any something wrong is a much more possible/honest mistake.